src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1231,6 +1231,11 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 						if (connectionMngInfo.serverInfo) {
 							connection.options.isCloud = connectionMngInfo.serverInfo.isCloud;
 						}
+						else if (connection.userName.length === 0) {
```

Comment:
Why are we not checking this on the STS side? The error should be coming from there, having to check here doesn't seem correct (what if there is a provider that accepts empty user names?)